### PR TITLE
auto-increment.md: add invalidate cache descriptions

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions))

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18543
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: auto-increment.md [hunk 1]
```diff
@@ -304,47 +304,108 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
+| t     | CREATE TABLE `t` (
+  `a` int(11) NOT NULL AUTO_INCREMENT,
+  PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin AUTO_INCREMENT=101 /*T![auto_id_cache] AUTO_ID_CACHE=100 */ |
++-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.00 sec)
 ```
 
-At this time, if you invalidate the auto-increment cache of this column and redo the implicit insertion, the result is as follows:
+At this time, if TiDB is restarted, the auto-increment ID cache will be lost, and new insert operations will allocate IDs starting from a higher value beyond the previously cached range.
```
Comment (qiancai) [inline auto-increment.md:334]:
```suggestion
At this time, if you restart TiDB, the auto-increment ID cache will be lost, and new insert operations will allocate IDs starting from a higher value beyond the previously cached range.
```

File: auto-increment.md [hunk 2]
```diff
@@ -304,47 +304,108 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
 
+### Clear the auto-increment ID cache
+
+In some scenarios, you might need to clear the auto-increment ID cache to ensure data consistency. For example:
+
+- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's writing operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
+- When your application involves both explicit ID insertion and implicit ID allocation, you need to clear the auto-increment ID cache to avoid conflicts between future implicitly allocated IDs and previously explicitly inserted IDs, which could result in primary key conflict errors. For more information, see [Uniqueness](/auto-increment.md#uniqueness).
+
+You can execute the `ALTER TABLE` statement with `AUTO_INCREMENT = 0` to clear the auto-increment ID cache on all TiDB nodes in the cluster. For example:
```
Comment (qiancai) [inline auto-increment.md:361]:
```suggestion
To clear the auto-increment ID cache on all TiDB nodes in the cluster, you can execute the `ALTER TABLE` statement with `AUTO_INCREMENT = 0`. For example:
```

File: auto-increment.md [hunk 3]
```diff
@@ -304,47 +304,108 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
 | a   |
 +-----+
+|   1 |
 | 101 |
 +-----+
-1 row in set (0.00 sec)
+2 rows in set (0.01 sec)
 ```
 
 The re-assigned value is `101`. This shows that the size of cache for allocating the auto-increment ID is `100`.
```
Comment (tangenta) [inline auto-increment.md:350]: re-assigned -> newly allocated?

File: auto-increment.md [hunk 4]
```diff
@@ -304,47 +304,108 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
 
 In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can be inserted properly.
 
+### Clear the auto-increment ID cache
```
Comment (tangenta) [inline auto-increment.md:354]:
```suggestion
### Invalidate ID cache
```

中文的 "无效化" 有些晦涩所以改成了 "清除"，这里英文还是保持 invalidate 比较合适？

**Review by tangenta — COMMENTED** [review]
Rest LGTM

File: auto-increment.md (hunk 3 above)
Comment (qiancai) [inline auto-increment.md:350]:
```suggestion
The newly allocated value is `101`. This shows that the size of cache for allocating auto-increment IDs is `100`.
```

File: auto-increment.md (hunk 4 above)
Comment (qiancai) [reply auto-increment.md:354]: 如果是这样，中文里 ”清除自增 ID 缓存“可以修改 ”使自增 ID 缓存失效“，因为”清除“和”使失效“含义还是有差别的

或者不确定是否可以用  ”重置自增 ID 缓存“ 来体现 ALTER TABLE t AUTO_INCREMENT = 0 对应的操作

**Review by tangenta — APPROVED** [review]

**Comment by qiancai** [comment]
@qiancai this is a test comment

**Comment by qiancai** [comment]
@qiancai This a another test comment.

**Comment by Glight29** [comment]
@qiancai  4test

File: auto-increment.md [hunk 5]
```diff
@@ -304,47 +304,108 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
 | 101 |
 +-----+
-1 row in set (0.00 sec)
+2 rows in set (0.01 sec)
 ```
 
-The re-assigned value is `101`. This shows that the size of cache for allocating the auto-increment ID is `100`.
+The newly allocated value is `101`. This shows that the size of cache for allocating auto-increment IDs is `100`.
 
 In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can be inserted properly.
```
Comment (hfxsd) [inline auto-increment.md:352]:
```suggestion
In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can insert data properly.
```

**Review by hfxsd — APPROVED** [review]

File: auto-increment.md [hunk 6]
```diff
@@ -304,46 +304,106 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
+The newly allocated value is `101`. This shows that the size of cache for allocating auto-increment IDs is `100`.
 
-In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can be inserted properly.
+In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can insert data properly.
+
+### Clear the auto-increment ID cache
+
+In some scenarios, you might need to clear the auto-increment ID cache to ensure data consistency. For example:
+
+- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's writing operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
```
Comment (qiancai) [inline auto-increment.md:357]:
```suggestion
- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
```

File: auto-increment.md [hunk 7]
```diff
@@ -304,46 +304,106 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
+The newly allocated value is `101`. This shows that the size of cache for allocating auto-increment IDs is `100`.
 
-In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can be inserted properly.
+In addition, when the length of consecutive IDs in a batch `INSERT` statement exceeds the length of `AUTO_ID_CACHE`, TiDB increases the cache size accordingly to ensure that the statement can insert data properly.
+
+### Clear the auto-increment ID cache
+
+In some scenarios, you might need to clear the auto-increment ID cache to ensure data consistency. For example:
+
+- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
```
Comment (qiancai) [inline auto-increment.md:357]:
```suggestion
<CustomContent platform="tidb">

- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.

</CustomContent>
<CustomContent platform="tidb-cloud">

- In the scenario of incremental replication using the [Data Migration](/tidb-cloud/migrate-incremental-data-from-mysql-using-data-migration.md) feature, once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.

</CustomContent>
```

File: auto-increment.md [hunk 8]
```diff
@@ -304,46 +304,115 @@ After the value `2030000` is inserted, the next value is `2060001`. This jump in
...
+<CustomContent platform="tidb">
+
+- In the scenario of incremental replication using [Data Migration (DM)](/dm/dm-overview.md), once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
+
+</CustomContent>
+<CustomContent platform="tidb-cloud">
+
+- In the scenario of incremental replication using the [Data Migration](/tidb-cloud/migrate-incremental-data-from-mysql-using-data-migration.md) feature, once the replication is complete, data writing to the downstream TiDB switches from DM to your application's write operations. Meanwhile, the ID writing mode of the auto-increment column usually switches from explicit insertion to implicit allocation.
+
+</CustomContent>
```
Comment (qiancai) [inline auto-increment.md:366]:
```suggestion
</CustomContent>

```

**Pull request merged**